validation/validator.test.js

```diff
@@ -0,0 +1,272 @@
+        finPeriode: 'aug',
+      },
+      sol: {
+        humus: ['frais', 'marécageux'],
+        tauxHumus: ['très faible', 'faible'],
+        ph: 3,
+      },
+      exposition: ['soleil', 'mi-ombre'],
+      nombrePlantations: 0,
+      utilisations: ['bac', 'massif'],
+      remarque: 'cette plante est fictionnelle',
+      representation: 'myLittleFougere.jpg',
+    };
+
+    const result = validator.validatePlante(invalidData);
+
+    assert.notEqual(result.errors.length, 0);
+  });
+
+  it('should not valid to low value for number', () => {
+    const invalidData = {
+      categorie: 'Fougère',
+      sousCategorie: 'persistant',
+      nomLatin: 'Fougus communum',
+      nom: 'Foougère commune',
+      cultivarPrincipal: 'alba',
+      cultivarSecondaires: ['bleu', 'rouge'],
+      indiceRusticite: 'normal',
+      envergure: 80,
+      hauteur: 60,
+      port: ['arque', 'étalé'],
+      feuillage: {
+        couleur: 'vert-gris',
+        resistanceFroid: 'persistant',
+        limbe: ['effilé', 'alterne'],
+      },
+      floraison: {
+        debutFloraison: 'apr',
+        finFloraison: 'jun',
+        couleur: ['lila', 'jaune'],
+      },
+      fruit: {
+        nom: 'baie',
+        debutPeriode: 'jul',
+        finPeriode: 'aug',
+      },
+      sol: {
+        humus: ['frais', 'marécageux'],
+        tauxHumus: ['très faible', 'faible'],
+        ph: 9,
+      },
+      exposition: ['soleil', 'mi-ombre'],
+      nombrePlantations: 3,
+      utilisations: ['bac', 'massif'],
+      remarque: 'cette plante est fictionnelle',
+      representation: 'myLittleFougere.jpg',
+    };
+
+    const result = validator.validatePlante(invalidData);
+
+    assert.notEqual(result.errors.length, 0);
+  });
+
+  it('should valid the given plante', () => {
+    const validData = {
+      categorie: 'Fougère',
+      sousCategorie: 'persistant',
+      nomLatin: 'Fougus communum',
+      nom: 'Foougère commune',
+      cultivarPrincipal: 'alba',
+      cultivarSecondaires: ['bleu', 'rouge'],
+      indiceRusticite: 'normal',
+      envergure: 80,
+      hauteur: 60,
+      port: ['arque', 'étalé'],
+      feuillage: {
+        couleur: 'vert-gris',
+        resistanceFroid: 'persistant',
+        limbe: ['effilé', 'alterne'],
+      },
+      floraison: {
+        debutFloraison: 'apr',
+        finFloraison: 'jun',
+        couleur: ['lila', 'jaune'],
+      },
+      fruit: {
+        nom: 'baie',
+        debutPeriode: 'jul',
+        finPeriode: 'aug',
+      },
+      sol: {
+        humus: ['frais', 'marécageux'],
+        tauxHumus: ['très faible', 'faible'],
+        ph: 6,
+      },
+      exposition: ['soleil', 'mi-ombre'],
+      nombrePlantations: 3,
+      utilisations: ['bac', 'massif'],
+      remarque: 'cette plante est fictionnelle',
+      representation: 'myLittleFougere.jpg',
+    };
+
+    const result = validator.validatePlante(validData);
+
+    assert.equal(result.errors.length, 0);
+  });
```

Comment:
Your tests are either too thorough / not thorough enough. The first test is good, the last one too, but the other seem to try to test the json schema library and not your code, the sentences in the `should` messages are a good indication of that.

You'd be better having only one test for invalid input, and check that the amount of errors is the one expected for example.

Reply:
Tests have been reduced to 3: first and last remain unchanged. Other tests have been replaced by only 1 checking this time on the number of errors generated.